File: 55-jump-game/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        minJump = 0
        for i in range(len(nums)-2,-1,-1):
            minJump += 1
            if nums[i]>=minJump:
                minJump = 0
            
        return minJump==0

File: 55-jump-game/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int ans = 0;
        
        for (int i=nums.size()-2;i>=0;i--){
            ans++;
            if (nums[i]>=ans){
                ans = 0;
            }
        }
        return ans==0;
    }
};